plugins/engines/ospray/OSPRayScene.cpp

```diff
@@ -84,7 +84,24 @@ void OSPRayScene::commit()
     commitTransferFunctionData();
 
     if (!rebuildScene && !addRemoveVolumes)
-        return;
+    {
+        // copy the list to avoid locking the mutex
+        bool doUpdate = false;
+        auto modelDescriptors = _modelDescriptors;
```

Comment:
If there's a race condition, the copy must still be thread-safe.

Reply:
? It's safe against add/remove of models?!